### DISCUSSION TITLE
Necessity to adjust schema for sqlite is a bug!

### DISCUSSION THREAD

**Discussion by jayarjo** [prisma/prisma#25823] [Q&A]
### Question

Before I finally encountered this I believed that prisma schema is universal for all supported engines, 'cause otherwise what's the point?! So I had the schema for postgres, quite rich and intricate, with types and enums (great feature) and then I thought maybe postgres is overkill, lets maybe try sqlite - after all switching should be a breeze! I was so wrong... In order to adapt for sqlite I had to strip all enums, convert fields to mere strings, strip varchars and json fields... why? why? It should be very easy on prisma side to simply ignore all of this or implement some support on Prisma level, for JSON fields for example? Now to switch to sqlite I will have to irreversibly screw up the schema, with no way to easily switch back if I change my mind. Also I'm not sure now if the syntax I used all over the codebase is actually portable...

**Comment by ludralph** [comment]
Hi @jayarjo 👋 

I understand your frustration with the differences between PostgreSQL and SQLite support in Prisma. You're right that there are some significant limitations when trying to switch between these database providers. I'll try to address some of the points you raised.

1. Enum support in SQLite: SQLite does not natively support enums. See the supported types [here](https://www.sqlite.org/datatype3.html). Prisma currently does not provide a polyfill for this feature in SQLite. This is a known limitation and there's an open feature [request](https://github.com/prisma/prisma/issues/2219) to add support for enums in SQLite.

2. JSON fields: SQLite does have native JSON support, but Prisma doesn't currently support the JSON field type for SQLite. There's an open [issue](https://github.com/prisma/prisma/issues/3786) for this as well.

Currently, there isn't a way to maintain a single schema that works identically for both PostgreSQL and SQLite. However, as a workaround, you could maintain separate schema files for different database providers and switch between them as needed. See this previous [comment](https://github.com/prisma/prisma/issues/2443#issuecomment-630679118) on how to do this.

**Reply by jayarjo** [reply]
Prisma could at least simply ignore enums and other fancy syntax, instead of demanding removal of those 🤷

**Reply by ludralph** [reply]
Hi @jayarjo 

I received confirmation from the team that both [Enum support](https://github.com/prisma/prisma/issues/2219) and [JSON fields](https://github.com/prisma/prisma/issues/3786) for SQLite are scheduled to be worked on this quarter. Keep an eye out for updates in the coming months!

**Reply by ludralph** [reply]
Hi @jayarjo 

We now have support for Json fields and Enums in SQLite. You can learn more in this [release note](https://github.com/prisma/prisma/releases/tag/6.2.0).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
